### PR TITLE
image-rs: add odd image test fixtures

### LINKED ISSUE #1258: image-rs | Add tests for odd images
There are some corner cases where image-rs once could not work but we introduce patches, we need a test suites to cover them everytime we introduce new patches.

- Duplicated layers: "gcr.io/k8s-staging-cloud-provider-ibm/ibm-vpc-block-csi-driver:master" (Just pull)
- whiteouts (Needs to build image and post-pulling check) https://github.com/confidential-containers/guest-components/issues/1210
```Dockerfile
FROM debian:stable-slim
RUN apt-get update && apt-get install -qy apache2
RUN rm -rf /etc/apache2
CMD ["find", "/etc/apache2"]
```
- whiteout issue II: "agnhost:2.21" (Just pull)

### PR DESCRIPTION
**Pull request by Xynnn007**
This also includes the image mentioned #1737 

Fixes #1258

### CODE REVIEW

**Comment by Xynnn007** [comment]
cc @stevenhorsman @mkulke  (Note that I changed the image ref)

**Review by mkulke — APPROVED** [review]
thanks!

File: image-rs/tests/odd_images.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    /// (deletion whiteout applied correctly).
+    OkPathAbsent(&'static str),
+    /// Pull must fail (e.g. empty local PAX rejected by astral-tokio-tar, #1737).
+    Err,
+}
+
+#[rstest::rstest]
+#[case::duplicated_layers("odd-duplicated-layers", Expectation::Ok)]
+#[case::whiteout_agnhost("odd-whiteout-agnhost", Expectation::Ok)]
+// This one is also cached inghcr.io/confidential-containers/guest-components/pathological-images/quay_io-curl-curl-8_4_0
```
Comment (stevenhorsman) [inline image-rs/tests/odd_images.rs]: nit: typo
```suggestion
// This one is also cached in ghcr.io/confidential-containers/guest-components/pathological-images/quay_io-curl-curl-8_4_0
```

**Review by stevenhorsman — APPROVED** [review]
LGTM apart from one typo I spotted. Thanks

**Review by fitzthum — APPROVED** [review]
